components/automate-cli/cmd/chef-automate/certRotate.go

```diff
@@ -486,6 +528,134 @@ func (c *certRotateFlow) getAllIPs(infra *AutomteHAInfraDetails) []string {
 	return ips
 }
 
+//getFilteredIps will return ips on which cert-rotation need to perform.
+func (c *certRotateFlow) getFilteredIps(serviceIps,skipIpsList []string) []string {
+	filteredIps := []string{}
+	for _, ip := range serviceIps {
+		if stringutils.SliceContains(skipIpsList, ip) == false {
+			filteredIps = append(filteredIps, ip)
+		}
+	}
+	return filteredIps
+}
+
+//compareCurrentCertsWithNewCerts compare current certs and new certs and returns ips to skip cert-rotation.
+func (c *certRotateFlow) compareCurrentCertsWithNewCerts(remoteService string, newCerts *certificates, flagsObj *certRotateFlags, currentCertsInfo *certShowCertificates) []string {
+	skipIpsList := []string{}
+	isCertsSame := true
+	if remoteService == CONST_AUTOMATE {
+		if flagsObj.node == "" {
+			isCertsSame = strings.TrimSpace(currentCertsInfo.AutomateRootCert) == newCerts.rootCA
+		}
+		skipIpsList = c.comparePublicCertAndPrivateCert(newCerts, currentCertsInfo.AutomateCertsByIP, isCertsSame, flagsObj)
+		return skipIpsList
+	}
+
+	if remoteService == CONST_CHEF_SERVER {
+		skipIpsList = c.comparePublicCertAndPrivateCert(newCerts, currentCertsInfo.ChefServerCertsByIP, isCertsSame, flagsObj)
+		return skipIpsList
+	}
+
+	if remoteService == CONST_OPENSEARCH {
+		if flagsObj.node == "" {
+			isCertsSame = strings.TrimSpace(currentCertsInfo.OpensearchRootCert) == newCerts.rootCA
+			isCertsSame = strings.TrimSpace(currentCertsInfo.OpensearchAdminCert) == newCerts.adminCert && isCertsSame
+			isCertsSame = strings.TrimSpace(currentCertsInfo.OpensearchAdminKey) == newCerts.adminKey && isCertsSame
+		}
+		skipIpsList = c.comparePublicCertAndPrivateCert(newCerts, currentCertsInfo.OpensearchCertsByIP, isCertsSame, flagsObj)
+		return skipIpsList
+	}
+
+	if remoteService == CONST_POSTGRESQL {
+		if flagsObj.node == "" {
+			isCertsSame = strings.TrimSpace(currentCertsInfo.PostgresqlRootCert) == newCerts.rootCA
+		}
+		skipIpsList = c.comparePublicCertAndPrivateCert(newCerts, currentCertsInfo.PostgresqlCertsByIP, isCertsSame, flagsObj)
+		return skipIpsList
+	}
+	return skipIpsList
+}
+
+//comparePublicCertAndPrivateCert compare new public-cert and private cert with current public-cert and private-cert and returns ips to skip cert-rotation.
+func (c *certRotateFlow) comparePublicCertAndPrivateCert(newCerts *certificates, certByIpList []CertByIP, isCertsSame bool, flagsObj *certRotateFlags) []string {
+	skipIpsList := []string{}
+	for _, currentCerts := range certByIpList {
+		isCertsSameTemp := false
+		if strings.TrimSpace(currentCerts.PublicKey) == newCerts.publicCert && strings.TrimSpace(currentCerts.PrivateKey) == newCerts.privateCert && isCertsSame {
```

Comment:
Add () for string comparison.
`if (strings.TrimSpace(currentCerts.PublicKey) == newCerts.publicCert) && (strings.TrimSpace(currentCerts.PrivateKey) == newCerts.privateCert) && isCertsSame {`